designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/ScenarioStatistics.scala

```diff
@@ -0,0 +1,178 @@
+      val attachmentsTotal          = sortedAttachmentCountList.sum
+      //        Comment stats
+      val comments        = listOfActivities.map(_.comments.length).sorted
+      val commentsTotal   = comments.sum
+      val commentsAverage = comments.sum / comments.length
+
+      Map(
+        AttachmentsAverage -> attachmentAverage,
+        AttachmentsTotal   -> attachmentsTotal,
+        CommentsTotal      -> commentsTotal,
+        CommentsAverage    -> commentsAverage
+      ).map { case (k, v) => (k.toString, v.toString) }
+    }
+  }
+
+  def getComponentStatistic(componentList: List[component.ComponentListElement]): Map[String, String] = {
+    if (componentList.isEmpty) {
+      Map.empty
+    } else {
+      val withoutFragments      = componentList.filterNot(comp => comp.componentType == ComponentType.Fragment)
+      val componentsByNameCount = withoutFragments.groupBy(_.name).size
+
+      Map(
+        ComponentsCount -> componentsByNameCount
+      ).map { case (k, v) => (k.toString, v.toString) }
+    }
+  }
+
+  private val flinkDeploymentManagerType = DeploymentManagerType("flinkStreaming")
+
+  private val liteK8sDeploymentManagerType = DeploymentManagerType("lite-k8s")
+
+  private val liteEmbeddedDeploymentManagerType = DeploymentManagerType("lite-embedded")
+
+  private val knownDeploymentManagerTypes =
+    Set(flinkDeploymentManagerType, liteK8sDeploymentManagerType, liteEmbeddedDeploymentManagerType)
+
+  // We have four dimensions:
+  // - scenario / fragment
+  // - processing mode: streaming, r-r, batch
+  // - dm type: flink, k8s, embedded, custom
+  // - status: active (running), other
+  // We have two options:
+  // 1. To aggregate statistics for every combination - it give us 3*4*2 + 3*4 = 36 parameters
+  // 2. To aggregate statistics for every dimension separately - it gives us 2+3+4+1 = 10 parameters
+  // We decided to pick the 2nd option which gives a reasonable balance between amount of collected data and insights
+  private[statistics] def determineStatisticsForScenario(inputData: ScenarioStatisticsInputData): Map[String, Int] = {
+    Map(
+      "s_s"     -> !inputData.isFragment,
+      "s_f"     -> inputData.isFragment,
+      "s_pm_s"  -> (inputData.processingMode == ProcessingMode.UnboundedStream),
+      "s_pm_b"  -> (inputData.processingMode == ProcessingMode.BoundedStream),
+      "s_pm_rr" -> (inputData.processingMode == ProcessingMode.RequestResponse),
+      "s_dm_f"  -> (inputData.deploymentManagerType == flinkDeploymentManagerType),
+      "s_dm_l"  -> (inputData.deploymentManagerType == liteK8sDeploymentManagerType),
+      "s_dm_e"  -> (inputData.deploymentManagerType == liteEmbeddedDeploymentManagerType),
+      "s_dm_c"  -> !knownDeploymentManagerTypes.contains(inputData.deploymentManagerType),
+      "s_a"     -> inputData.status.contains(SimpleStateStatus.Running),
+    ).mapValuesNow(if (_) 1 else 0)
+  }
+
+  private def calculateMedian[T: Numeric](list: List[T]): T = {
```

Comment:
I will fell calm if this function could handle empty list too :). Maybe we should extract analogous function for the average? Additionally I would extract min and max functions which would handle empty list

Reply:
And currently it doesn't? `Get` gives `Some(value)` if given index exists in a list or `None` if it's outside the scope. Then `getOrElse` transforms `Option` to either value or zero.
At least that's my understanding but may be wrong

Reply:
Sorry my bad, it's safe :)

Reply:
I remember why I wrote it. I saw and average calculation and I generalised it to the median. So other remarks I would still consider to do :)

Reply:
```
def calculateAverage[T: Numeric](list: List[T]): T = {
  val result = implicitly[Numeric[T]].toInt(list.sum) / list.length
  implicitly[Numeric[T]].fromInt(result)
}

calculateAverage(List[Int]())
```
```
def calculateAverage[T](list: List[T])(implicit evidence$1: Numeric[T]): T

java.lang.ArithmeticException: / by zero
  at calculateAverage(<console>:2)
  ... 41 elided
```